### PR TITLE
[STRAITSX-47060] hero NFT

### PR DESCRIPTION
**Pull request by jacobshan**
[Context]
- ERC1155 hero NFT 
- make base URI upgradable
- only owner and whitelisted address(PBM contract in pilot6 ) can mint hero NFT

[Note]
- token id is unique to each merchant.
- image url in the metadata should be provided by Grab
- metadata json better host on a dedicated github/s3

### CODE REVIEW

File: heroNFT2023SEP/contracts/HeroNFT.sol [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+import "@openzeppelin/contracts/token/ERC1155/extensions/ERC1155Burnable.sol";
+import "@openzeppelin/contracts/token/ERC1155/extensions/ERC1155Supply.sol";
+import "@openzeppelin/contracts/utils/Strings.sol";
+
+contract HeroNFT is ERC1155, Ownable, Pausable, ERC1155Burnable, ERC1155Supply {
+    constructor() ERC1155("") {}
+
+    // better open a dedicated repo on github to store the metadata
+    string baseURI =
+        "https://raw.githubusercontent.com/StraitsX/Orchid-PBM/STRAITSX-4706/commemorative_nft/heroNFT2023SEP/metadata/";
```
Comment (alcedo) [inline heroNFT2023SEP/contracts/HeroNFT.sol:16]: created one here https://github.com/StraitsX/NFT/blob/main/README.md#introduction

File: heroNFT2023SEP/metadata/0.json [hunk 2]
```diff
@@ -0,0 +1 @@
+{"description": "cute dog", "name": "SUSHI THE SHIBA", "attributes": [{"trait_type": "Base", "value": "Canine"}], "image": "https://fazzfinancialgroup.notion.site/image/https%3A%2F%2Fs3-us-west-2.amazonaws.com%2Fsecure.notion-static.com%2F3632850a-85ff-418b-a399-9fe6cf6ab9ce%2FUntitled.png?id=2df5203c-ca21-4a09-8395-e624e6d3b259&table=block&spaceId=bdc391a7-c326-437a-bb1e-2ce2ad0e7da0&width=1720&userId=&cache=v2" }
```
Comment (alcedo) [inline heroNFT2023SEP/metadata/0.json:1]: can be moved also  https://github.com/StraitsX/NFT/blob/main/README.md#introduction

Comment (alcedo) [reply heroNFT2023SEP/metadata/0.json:1]: Can change to this also

{
   "description":"Hero NFT",
   "name":"SG Hero Merchant series 2023",
   "attributes":[
      {
         "trait_type":"Base",
         "value":"Cultural - Food"
      }
   ],
   "image":"https://fazzfinancialgroup.notion.site/image/https%3A%2F%2Fs3-us-west-2.amazonaws.com%2Fsecure.notion-static.com%2F3632850a-85ff-418b-a399-9fe6cf6ab9ce%2FUntitled.png?id=2df5203c-ca21-4a09-8395-e624e6d3b259&table=block&spaceId=bdc391a7-c326-437a-bb1e-2ce2ad0e7da0&width=1720&userId=&cache=v2"
}

File: heroNFT2023SEP/scripts/deploy.js [hunk 3]
```diff
@@ -0,0 +1,14 @@
+// We require the Hardhat Runtime Environment explicitly here. This is optional
+// but useful for running the script in a standalone fashion through `node <script>`.
+//
+// You can also run a script with `npx hardhat run <script>`. If you do that, Hardhat
+// will compile your contracts, add the Hardhat Runtime Environment's members to the
+// global scope, and execute the script.
+const hre = require('hardhat');
+
+async function main() {
+  const heroNFT = await hre.ethers.deployContract('HeroNFT');
```
Comment (alcedo) [inline heroNFT2023SEP/scripts/deploy.js:10]: include the script to do minting and mint to address?

**Review by alcedo — CHANGES_REQUESTED** [review]

File: heroNFT2023SEP/contracts/HeroNFT.sol [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+import "@openzeppelin/contracts/security/Pausable.sol";
+import "@openzeppelin/contracts/token/ERC1155/extensions/ERC1155Burnable.sol";
+import "@openzeppelin/contracts/token/ERC1155/extensions/ERC1155Supply.sol";
+import "@openzeppelin/contracts/utils/Strings.sol";
+
+contract HeroNFT is ERC1155, Ownable, Pausable, ERC1155Burnable, ERC1155Supply {
+    constructor() ERC1155("") {}
+
+    // better open a dedicated repo on github to store the metadata
+    string baseURI = "https://raw.githubusercontent.com/StraitsX/NFT/main/heroNFT2023SEP/heroNFT2023SEP/";
```
Comment (jacobshan) [inline heroNFT2023SEP/contracts/HeroNFT.sol:15]: don't have access to push PR to the repo yet. once it's pushed and merged this link will be working.

File: heroNFT2023SEP/README.md [hunk 5]
```diff
@@ -1,8 +1,6 @@
-# Sample Hardhat Project
-
-This project demonstrates a basic Hardhat use case. It comes with a sample contract, a test for that contract, and a script that deploys that contract.
-
-Try running some of the following tasks:
+# heroNFT2023SEP
+Commemorative NFT for Singapore Grand Prix PBM Pilot 2023 Sep. 
+This ERC1155 behaves like a ERC721, ensuring (address, token_id) is unique and only 1 of such pair can ever exist in the smart contract.
```
Comment (alcedo) [inline heroNFT2023SEP/README.md:3]: <a href="https://gitme.me/image?url=https%3A%2F%2Fmedia3.giphy.com%2Fmedia%2FpCO5tKdP22RC8%2Fgiphy.gif&token=nice" data-gitmeme-token="nice"></a>

File: heroNFT2023SEP/contracts/HeroNFT.sol (hunk 4 above)
Comment (alcedo) [reply heroNFT2023SEP/contracts/HeroNFT.sol:15]: <a href="https://gitme.me/image?url=https%3A%2F%2Fmedia3.giphy.com%2Fmedia%2F3oz8xQQP4ahKiyuxHy%2Fgiphy.gif&token=ok" data-gitmeme-token="ok"></a>

**Review by alcedo — APPROVED** [review]
<a href="https://gitme.me/image?url=https%3A%2F%2Fmedia4.giphy.com%2Fmedia%2FHw8vYF4DNRCKY%2Fgiphy.gif&token=ship" data-gitmeme-token="ship"></a> it

**Comment by alcedo** [comment]
image

**Pull request merged**